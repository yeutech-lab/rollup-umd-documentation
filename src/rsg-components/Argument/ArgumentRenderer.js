import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import Markdown from 'rsg-components/Markdown';
import Name from 'rsg-components/Name';
import Type from 'rsg-components/Type';
import Group from 'react-group';

export const styles = ({ space }) => ({
	block: {
		marginBottom: space[2],
	},
});

export function ArgumentRenderer({ classes, name, type, description, returns, block, ...props }) {
	return (
		<Group className={block && classes.block} {...props}>
			{returns && 'Returns'}
			{name && (
				<span>
					<Name>{name}</Name>
					{type && ':'}
				</span>
			)}
			{type && <Type>{type.name}</Type>}
			{type && description && ` â€” `}
			{description && <Markdown text={`${description}`} inline />}
		</Group>
	);
}

ArgumentRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	name: PropTypes.string,
	type: PropTypes.object,
	description: PropTypes.string,
	returns: PropTypes.bool,
	block: PropTypes.bool,
};

export default Styled(styles)(ArgumentRenderer);
