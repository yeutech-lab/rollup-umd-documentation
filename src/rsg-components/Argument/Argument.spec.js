import React from 'react';
import { ArgumentRenderer, styles } from './ArgumentRenderer';

const name = 'Foo';
const type = { name: 'Array' };
const description = 'Converts foo to bar';
const props = {
	classes: classes(styles),
};

it('should render argument', () => {
	const actual = shallow(
		<ArgumentRenderer {...props} name={name} type={type} description={description} />
	);

	expect(actual).toMatchSnapshot();
});

it('should render argument without type', () => {
	const actual = shallow(<ArgumentRenderer {...props} name={name} description={description} />);

	expect(actual).toMatchSnapshot();
});

it('should render argument without description', () => {
	const actual = shallow(<ArgumentRenderer {...props} name={name} type={type} />);

	expect(actual).toMatchSnapshot();
});

it('should render return value', () => {
	const actual = shallow(
		<ArgumentRenderer {...props} type={type} description={description} returns />
	);

	expect(actual).toMatchSnapshot();
});

it('should render with block styles', () => {
	const actual = shallow(<ArgumentRenderer {...props} block />);

	expect(actual).toMatchSnapshot();
});
