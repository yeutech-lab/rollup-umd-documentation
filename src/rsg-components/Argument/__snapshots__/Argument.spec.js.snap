// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render argument 1`] = `
<Group
  separator=" "
>
  <span>
    <Styled(Name)>
      Foo
    </Styled(Name)>
    :
  </span>
  <Styled(Type)>
    Array
  </Styled(Type)>
   — 
  <Markdown
    inline={true}
    text="Converts foo to bar"
  />
</Group>
`;

exports[`should render argument without description 1`] = `
<Group
  separator=" "
>
  <span>
    <Styled(Name)>
      Foo
    </Styled(Name)>
    :
  </span>
  <Styled(Type)>
    Array
  </Styled(Type)>
</Group>
`;

exports[`should render argument without type 1`] = `
<Group
  separator=" "
>
  <span>
    <Styled(Name)>
      Foo
    </Styled(Name)>
  </span>
  <Markdown
    inline={true}
    text="Converts foo to bar"
  />
</Group>
`;

exports[`should render return value 1`] = `
<Group
  separator=" "
>
  Returns
  <Styled(Type)>
    Array
  </Styled(Type)>
   — 
  <Markdown
    inline={true}
    text="Converts foo to bar"
  />
</Group>
`;

exports[`should render with block styles 1`] = `
<Group
  className="block"
  separator=" "
/>
`;
